src/targets/gpu/fuse_mlir.cpp

```diff
@@ -1021,6 +1027,95 @@ struct find_unpack_int4_mlir_op
     }
 };
 
+/**
+ * Move unpack_fp4 and reshape/slice instructions into the mlir_op submodule.
+ * rocMLIR will do the unpacking and dequantization.
+ */
+struct find_unpack_fp4_mlir_op
+{
+    const std::unordered_set<std::string> skip_set = {"reshape", "slice"};
+
+    auto matcher() const
+    {
+        auto skip_reshapes_to_unpack =
+            match::skip(match::name(skip_set))(match::name("unpack_fp4"));
```

Comment:
This shouldn't be needed as reshape and slice already get fused. You just need to look for `unpack_fp4`.